### ISSUE TITLE
search: introduce searchable fields

### ISSUE THREAD

**Issue by tiborsimko** [cernopendata/opendata.cern.ch#1738]
On COD2, we have been having many searchable fields, such as author, title, collection, etc. People could do various advanced Boolean queries with them.  

On COD3, some fields can be searched in a fielded way already, for example:

- title:muon

works, but some useful fields cannot be searched in a fielded way, for example:

- collection:CMS-Primary-Datasets
- type:Datasets
- collision_type:PbPb
- year:2011

(1) The goal of this issue is to clarify which fields would be interesting to offer for such a fielded search, so that people could do various targeted queries if they want.  (and not only free text style "find-this-anywhere" queries)

(2) We should obviously offer the common fields that say e.g. Google Scholar offers, such as author, title, year. Because people are accustomed to those. Plus we should offer some COD-specific ones such as collection, type, experiment, collaboration, etc.

(3) I would propose to make every faceted field to be *also* searchable in a fielded way in the query syntax (`field:value`), so that if the UI does not permit an easy combination of AND/OR between facets and between values within facets, we would still be able to recommend to power users to write stuff like: *(just an example)*

- (experiment:cms OR experiment:atlas) AND type:datasets AND energy:7TeV 
- type:glossary +muon
- ...

(3) One could even imagine offering a full JSON structure search like:

- type.primary:Datasets and usage.description:test

but that might be too futuristic 😄 A simple list of 6-7 most useful fields might be enough in my eyes.

Most people will seach "anywhere" by default anyway...

CC @daslerr @katilp @sefeg

**Comment by daslerr** [comment]
I agree that offering a full JSON structure search might be too much. I think option 3 where every faceted field is also available as a fielded search seems like the best way to go, if it's not too much trouble to implement. I wouldn't think it's high priority, though, since it's mostly aimed at power users.

**Comment by suenjedt** [comment]
While I agree with @daslerr and @tiborsimko on a preferred option 3, I suggest that @sefeg compiles the list of 6-7 things and we take it from there, given the time pressure.

**Comment by sefeg** [comment]
I propose to consider the following for the moment:

- title
- type
- experiment
- year
- file_type
- collision_energy
- topic_category
- keywords

The list is basically a reflection of our facet cateories
